express-zod-api/src/documentation-helpers.ts

```diff
@@ -498,42 +568,42 @@ const makeNullableType = ({
   return type ? [...new Set(type).add("null")] : "null";
 };
 
-export const depictEffect: Depicter = (
-  schema: z.ZodEffects<z.ZodTypeAny>,
+export const depictPipeline: Depicter = (
+  { _zod: { def } }: $ZodPipe,
   { isResponse, next },
 ) => {
-  const input = next(schema.innerType());
-  const { effect } = schema._def;
-  if (isResponse && effect.type === "transform" && isSchemaObject(input)) {
-    const outputType = getTransformedType(schema, makeSample(input));
-    if (outputType && ["number", "string", "boolean"].includes(outputType))
-      return { type: outputType as "number" | "string" | "boolean" };
-    else return next(z.any());
-  }
-  if (!isResponse && effect.type === "preprocess" && isSchemaObject(input)) {
-    const { type: inputType, ...rest } = input;
-    return { ...rest, format: `${rest.format || inputType} (preprocessed)` };
+  const target = def[isResponse ? "out" : "in"];
+  const opposite = def[isResponse ? "in" : "out"];
+  if (target._zod.def.type === "transform") {
+    const opposingDepiction = next(opposite);
+    if (isSchemaObject(opposingDepiction)) {
+      if (!isResponse) {
+        const { type: opposingType, ...rest } = opposingDepiction;
+        return {
+          ...rest,
+          format: `${rest.format || opposingType} (preprocessed)`,
+        };
+      } else {
+        const targetType = getTransformedType(
+          target as z.ZodTransform, // @todo should use $ZodTransform from core?
+          makeSample(opposingDepiction),
+        );
+        if (targetType && ["number", "string", "boolean"].includes(targetType))
+          return { type: targetType as "number" | "string" | "boolean" };
+        else return next(z.any());
+      }
+    }
   }
-  return input;
+  return next(target);
 };
 
-export const depictPipeline: Depicter = (
-  { _def }: z.ZodPipeline<z.ZodTypeAny, z.ZodTypeAny>,
-  { isResponse, next },
-) => next(_def[isResponse ? "out" : "in"]);
-
-export const depictBranded: Depicter = (
-  schema: z.ZodBranded<z.ZodTypeAny, string | number | symbol>,
-  { next },
-) => next(schema.unwrap());
-
 export const depictLazy: Depicter = (
-  lazy: z.ZodLazy<z.ZodTypeAny>,
+  lazy: $ZodLazy,
   { next, makeRef },
-): ReferenceObject => makeRef(lazy, () => next(lazy.schema));
+): ReferenceObject => makeRef(lazy, () => next(lazy._zod.def.getter()));
```

Comment:
this does no longer work, because `getter()` returns new instance each time, so that `WeakSet` does not match them

Reply:
fixed